### PR TITLE
Snapshot of APIs for 12.3.

### PR DESCRIPTION
**Pull request by neilcsmith-net**
Snapshot of APIs for review leading up to 12.3 feature freeze (Jan ~22nd). Currently taken from state of master.  I'll keep this PR up-to-date with the release branch as we move through the release process, so we can keep review in one place, and merge into master once the release is out.

Please add additional reviewers as necessary.

### CODE REVIEW

**Review by geertjanw — APPROVED** [review]
Looks good to me.

**Comment by lkishalmi** [comment]
Scrolled it through. Just would like to mention that my changes were intentional. As I've checked others make sense as well. The only thing just popped out was the jaxb changes, though I can't really judge what was happening there at the first sight.

**Comment by matthiasblaesing** [comment]
JAXB was updated, so it is expected.

File: ide/projectapi/nbproject/org-netbeans-modules-projectapi.sig [hunk 1]
```diff
@@ -117,6 +117,10 @@ meth public static java.util.Set<org.netbeans.api.project.Project> getDependency
 meth public static java.util.prefs.Preferences getPreferences(org.netbeans.api.project.Project,java.lang.Class,boolean)
  anno 1 org.netbeans.api.annotations.common.NonNull()
  anno 2 org.netbeans.api.annotations.common.NonNull()
+meth public static org.netbeans.api.project.Project parentOf(org.netbeans.api.project.Project)
```
Comment (JaroslavTulach) [inline ide/projectapi/nbproject/org-netbeans-modules-projectapi.sig:120]: I see. Nicely generalized API originating from Gradle.

File: java/java.lsp.server/nbproject/org-netbeans-modules-java-lsp-server.sig [hunk 2]
```diff
@@ -63,6 +63,17 @@ meth public final void setStatusText(java.lang.String)
 supr org.openide.awt.StatusDisplayer
 hfds text
 
+CLSS public abstract org.netbeans.modules.java.lsp.server.ui.AbstractProgressEnvironment
```
Comment (JaroslavTulach) [inline java/java.lsp.server/nbproject/org-netbeans-modules-java-lsp-server.sig:66]: OK.

File: platform/api.progress.nb/nbproject/org-netbeans-api-progress-nb.sig [hunk 3]
```diff
@@ -158,6 +171,8 @@ cons public init()
 meth protected static org.netbeans.api.progress.ProgressHandle getProgressHandle(org.netbeans.api.progress.aggregate.AggregateProgressHandle)
 meth protected static org.netbeans.api.progress.aggregate.AggregateProgressHandle doCreateHandle(java.lang.String,org.netbeans.api.progress.aggregate.ProgressContributor[],org.openide.util.Cancellable,boolean,org.netbeans.api.progress.ProgressHandle)
 meth public static org.netbeans.api.progress.aggregate.AggregateProgressHandle createHandle(java.lang.String,org.netbeans.api.progress.aggregate.ProgressContributor[],org.openide.util.Cancellable)
+meth public static org.netbeans.api.progress.aggregate.AggregateProgressHandle createHandle(java.lang.String,org.netbeans.api.progress.aggregate.ProgressContributor[],org.openide.util.Cancellable,javax.swing.Action)
+meth public static org.netbeans.api.progress.aggregate.AggregateProgressHandle createSystemHandle(java.lang.String,org.netbeans.api.progress.aggregate.ProgressContributor[],org.openide.util.Cancellable,javax.swing.Action)
```
Comment (JaroslavTulach) [inline platform/api.progress.nb/nbproject/org-netbeans-api-progress-nb.sig:175]: OK.

**Review by JaroslavTulach — APPROVED** [review]

File: platform/api.progress.compat8/nbproject/org-netbeans-api-progress-compat8.sig [hunk 4]
```diff
@@ -70,18 +70,21 @@ fld public final static int STATE_FINISHED = 2
 fld public final static int STATE_INITIALIZED = 0
 fld public final static int STATE_REQUEST_STOP = 3
 fld public final static int STATE_RUNNING = 1
+meth protected final void markCustomPlaced()
 meth protected final void setController(org.netbeans.modules.progress.spi.Controller)
 meth public boolean isAllowCancel()
 meth public boolean isAllowView()
 meth public boolean isCustomPlaced()
 meth public boolean isInSleepMode()
+meth public boolean requestAction(java.lang.String,javax.swing.Action)
```
Comment (sdedic) [inline platform/api.progress.compat8/nbproject/org-netbeans-api-progress-compat8.sig:79]: OK/

File: platform/api.progress.nb/nbproject/org-netbeans-api-progress-nb.sig [hunk 5]
```diff
@@ -87,7 +89,9 @@ meth public final void switchToDeterminate(int,long)
 meth public final void switchToIndeterminate()
 meth public static org.netbeans.api.progress.ProgressHandle createHandle(java.lang.String)
 meth public static org.netbeans.api.progress.ProgressHandle createHandle(java.lang.String,org.openide.util.Cancellable)
+meth public static org.netbeans.api.progress.ProgressHandle createHandle(java.lang.String,org.openide.util.Cancellable,javax.swing.Action)
```
Comment (sdedic) [inline platform/api.progress.nb/nbproject/org-netbeans-api-progress-nb.sig:92]: OK.

**Review by sdedic — APPROVED** [review]

**Comment by neilcsmith-net** [comment]
Updated these based on current state of release branch (beta2). FlatLaf conflict is interesting - due to JDK version in generating sigfile I assume?

I will do a force push or new PR from scratch to check and clean this up before merging, but thought it useful to have a diff as we update for now.

**Comment by neilcsmith-net** [comment]
Closing this in favour of new PR for merging.

**Pull request closed without merging**